`catching polymorphic type` warnings with gcc 8.1

See https://github.com/openzim/libzim/issues/154

```
[5/6] Compiling C++ object 'src/src@@zimwriterfs@exe/tools.cpp.o'.                                     
../src/tools.cpp: In function ‘std::__cxx11::string getMimeTypeForFile(const string&)’:                
../src/tools.cpp:493:19: warning: catching polymorphic type ‘class std::out_of_range’ by value [-Wcatch-value=]                                                                                                
     } catch (std::out_of_range) {}                                                                    
                   ^~~~~~~~~~~~                                                                        
../src/tools.cpp:499:17: warning: catching polymorphic type ‘class std::out_of_range’ by value [-Wcatch-value=]                                                                                                
   } catch (std::out_of_range) {}                                                                      
                 ^~~~~~~~~~~~             
```

--- a/src/tools.cpp
+++ b/src/tools.cpp
@@ -490,13 +490,13 @@ std::string getMimeTypeForFile(const std::string& filename)
     mimeType = filename.substr(index_of_last_dot + 1);
     try {
       return extMimeTypes.at(mimeType);
-    } catch (std::out_of_range) {}
+    } catch (std::out_of_range&) {}
   }
 
   /* Try to get the mimeType from the cache */
   try {
     return fileMimeTypes.at(filename);
-  } catch (std::out_of_range) {}
+  } catch (std::out_of_range&) {}
 
   /* Try to get the mimeType with libmagic */
   try {

